### ISSUE TITLE
[Gazebo9][Windows] ModelDatabase::GetModelPath implementation is incomplete on Windows build.

### ISSUE THREAD

**Issue by osrf-migration** [osrf/gazebo#2611]
**Original report ([archived issue](https://osrf-migration.github.io/gazebo-gh-pages/#!/osrf/gazebo/issues/2611)) by Sean Yen (Bitbucket: [seanyen-msft](https://bitbucket.org/%7B9f25a1e7-e37c-4707-a084-64e8be6f3785%7D/)).**

----------------------------------------
https://github.com/osrf/gazebo/blob/ab82c04da4afe36cb7e6b9842a1d7b4cdfa35570/gazebo/common/ModelDatabase.cc#L506

I noticed that the GetModelPath requires libtar and maybe because of the lack of libtar port on Windows, some code path is disabled on Windows build, which breaks the scenario to use online models.

I'd like to use this bug to track how we should address this for Windows build.

**Comment by traversaro** [comment]
While using `libarchive`  is probably a cleaner version, @seanyen solution of using the `tar` command on Windows added in https://github.com/osrf/gazebo/pull/2789/files#diff-8f453ff422db23b46311978f86b355af2060a7833e4201045d5112fd2d502990 seems to be working fine, so it probably make sense to propagate it to Gazebo 11 as well (were it is missing, see https://github.com/osrf/gazebo/blob/gazebo11_11.5.0/gazebo/common/ModelDatabase.cc#L506 that is missing the `#else` block w.r.t. to https://github.com/osrf/gazebo/blob/gazebo9_9.17.0/gazebo/common/ModelDatabase.cc#L509 . 

TL;DR: To solve this, the simplest solution is to forward propagate https://github.com/osrf/gazebo/pull/2789 to the `gazebo11` branch.
